R/futuremice.R

```diff
@@ -87,57 +94,71 @@ futuremice <- function(data, m = 5, seed = NA, parallelseed = NA, n.core = NULL,
   # number of available cores
   available <- parallelly::availableCores(logical = use.logical)
   
-  # check if arguments match CPU specifications
   if (!is.null(n.core)) {
+    # check if arguments match CPU specifications
     if (n.core > available) {
-      stop("Number of cores specified is greater than the number of logical cores in your CPU")
+      warning(paste("Number of cores specified is greater than the number of logical cores in your CPU, futuremice falls back to", available, "cores."))
+      n.core <- available
+    } 
+    if (n.core > m) {
+      warning(paste("The number of cores exceeds the number of imputations. The number of cores used is set equal to the number of imputations (m =", m,")."))
+      n.core <- m
     }
+  } else {
+    n.core <- available - 1
+    message(paste("Number of cores not specified. Based on your machine a value of n.core =", n.core, "is chosen; the imputations are distributed about equally over the cores."))
   }
-
-  # determine course of action when not all arguments specified
-  if (!is.null(n.core) & is.null(n.imp.core)) {
-    n.imp.core <- m
-    warning(paste("Number of imputations per core not specified: n.imp.core = m =", m, "has been used"))
-  }
-  if (is.null(n.core) & !is.null(n.imp.core)) {
-    n.core <- available - 1 # leave one out by default for other processes. users can override if wanted
-    warning(paste("Number of cores not specified. Based on your machine a value of n.core =", parallelly::availableCores(logical = TRUE) - 1, "is chosen"))
-  }
-  if (is.null(n.core) & is.null(n.imp.core)) {
-    specs <- match.cluster(n.core = (available - 1), m = m)
-    n.core <- specs$cores
-    n.imp.core <- specs$imps
+  
+  if (n.core > 1) {
+    dist.core <- cut(1:m, n.core, labels = paste0("core", 1:n.core))
+  } else {
+    dist.core <- rep("core1", m) 
   }
+  
+  n.imp.core <- as.vector(table(dist.core))
+  
   if (!is.na(seed)) {
     if (n.core > 1) {
-      warning("Be careful; the specified seed is equal for all imputations. Please consider specifying parallelseed instead.")
+      if (interactive()) {
```

Comment:
If the user is in an interactive session, ask whether the seed was specified on purpose. If this was the case, continue but still output a warning about potential unintended consequences. If the seed was specified incorrectly, inform the user that `seed` is set to `NA`, but that the `parallelseed` is set for the user, and accessible in the output object. Otherwise, the user can stop the procedure and specify the seed manually, or simply use mice or whatever.